editor/components/block-list/with-hover-areas.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * WordPress dependencies
+ */
+import { Component, findDOMNode } from '@wordpress/element';
+
+const withHoverAreas = ( WrappedComponent ) => {
+	class WithHoverAreasComponent extends Component {
+		constructor() {
+			super( ...arguments );
+			this.state = {
+				hoverArea: null,
+			};
+			this.onMouseLeave = this.onMouseLeave.bind( this );
+			this.onMouseMove = this.onMouseMove.bind( this );
+		}
+
+		componentDidMount() {
+			// Disable reason: We use findDOMNode to avoid unnecessary extra dom Nodes
+			// eslint-disable-next-line react/no-find-dom-node
+			this.container = findDOMNode( this );
+			this.container.addEventListener( 'mousemove', this.onMouseMove );
+			this.container.addEventListener( 'mouseleave', this.onMouseLeave );
+		}
+
+		componentWillUnmount() {
+			this.container.removeEventListener( 'mousemove', this.onMouseMove );
+			this.container.removeEventListener( 'mouseleave', this.onMouseLeave );
+		}
+
+		onMouseLeave() {
+			if ( this.state.hoverArea ) {
+				this.setState( { hoverArea: null } );
+			}
+		}
+
+		onMouseMove( event ) {
+			const { width, left, right } = this.container.getBoundingClientRect();
```

Comment:
>## What forces layout / reflow
>
>_[...]_ This is also called reflow or layout thrashing, and is common performance bottleneck.
>
>**Box metrics**
>
>- _[...]_ `elem.getBoundingClientRect()`

https://gist.github.com/paulirish/5d52fb081b3570c81e3a

And in a `mousemove`, it's likely being called hundreds of times 🙁

Reply:
I guess I can look for alternatives to `getBoundingClientRect`

Reply:
Can't think of any alternative without browser reflow